zio-schema-thrift/shared/src/main/scala/zio/schema/codec/ThriftCodec.scala

```diff
@@ -0,0 +1,1327 @@
+import zio.{Chunk, ChunkBuilder, ZIO}
+import zio.schema.{Schema, StandardType}
+import zio.stream.ZTransducer
+
+import java.nio.ByteBuffer
+import java.time.{DayOfWeek, Duration, Instant, LocalDate, LocalDateTime, LocalTime, Month, MonthDay, OffsetDateTime, OffsetTime, Period, Year, YearMonth, ZoneId, ZoneOffset, ZonedDateTime}
+import java.util.UUID
+import scala.annotation.tailrec
+import scala.collection.immutable.ListMap
+import scala.util.{Failure, Success, Try}
+import scala.util.control.NonFatal
+
+object ThriftCodec extends Codec {
+  override def encoder[A](schema: Schema[A]): ZTransducer[Any, Nothing, A, Byte] =
+    ZTransducer.fromPush(
+      (opt: Option[Chunk[A]]) =>
+        ZIO.succeed(opt.map(values => values.flatMap(new Encoder().encode(schema, _))).getOrElse(Chunk.empty))
+    )
+
+  override def decoder[A](schema: Schema[A]): ZTransducer[Any, String, Byte, A] =
+    ZTransducer.fromPush(
+      (opt: Option[Chunk[Byte]]) =>
+        ZIO.fromEither(
+          opt
+            .map(
+              chunk =>
+                new Decoder(chunk)
+                  .decode(Chunk.empty, schema)
+                  .map(Chunk(_))
+                  .left
+                  .map(err => s"Error at path /${err.path.mkString(".")}: ${err.error}")
+            )
+            .getOrElse(Right(Chunk.empty))
+        )
+    )
+
+  override def encode[A](schema: Schema[A]): A => Chunk[Byte] = a => new Encoder().encode(schema, a)
+
+  override def decode[A](schema: Schema[A]): Chunk[Byte] => Either[String, A] =
+    ch =>
+      if (ch.isEmpty)
+        Left("No bytes to decode")
+      else
+        new Decoder(ch)
+          .decode(Chunk.empty, schema)
+          .left
+          .map(
+            err => s"Error at path /${err.path.mkString(".")}: ${err.error}"
+          )
+
+  object Thrift {
+
+    def singleSchema[A](codec: Schema[A]): Schema[ListMap[String, _]] = Schema.record(Schema.Field("value", codec))
+
+    def monthDayStructure(): Seq[Schema.Field[Int]] =
+      Seq(
+        Schema.Field("month", Schema.Primitive(StandardType.IntType)),
+        Schema.Field("day", Schema.Primitive(StandardType.IntType))
+      )
+
+    def periodStructure(): Seq[Schema.Field[Int]] = Seq(
+      Schema.Field("years", Schema.Primitive(StandardType.IntType)),
+      Schema.Field("months", Schema.Primitive(StandardType.IntType)),
+      Schema.Field("days", Schema.Primitive(StandardType.IntType))
+    )
+
+    def yearMonthStructure(): Seq[Schema.Field[Int]] =
+      Seq(
+        Schema.Field("year", Schema.Primitive(StandardType.IntType)),
+        Schema.Field("month", Schema.Primitive(StandardType.IntType))
+      )
+
+    def durationStructure(): Seq[Schema.Field[_]] =
+      Seq(
+        Schema.Field("seconds", Schema.Primitive(StandardType.LongType)),
+        Schema.Field("nanos", Schema.Primitive(StandardType.IntType))
+      )
+  }
+
+  private def unwrapLazy(schema: Schema[_]) = schema match {
+    case Schema.Lazy(s) =>
+      s()
```

Comment:
nit: In `Schema.Lazy` we assign the lazily derived `Schema` in a val for access without invoking the function again, so this can be changed to 

```case s @ Schema.Lazy(_) => s.schema```

Reply:
Great. Changed